src/handlers.rs

```diff
@@ -1,21 +1,17 @@
 //! API Handlers
-use actix::{ActorResponse, Addr};
 use actix_web::{
-    error, AsyncResponder, Error, FromRequest, FutureResponse, HttpRequest, HttpResponse, Json,
-    Path, Query, Responder, State,
+    Error, FromRequest, FutureResponse, HttpRequest, HttpResponse, Json, Path, Query, State,
 };
-use futures::Future;
+use futures::future::result;
 // Hawk lib brings in some libs that don't compile at the moment for some reason
 //use hawk::
 use serde::de::{Deserialize, Deserializer};
 
-use dispatcher;
+use data;
 
 /// This is the global HTTP state object that will be made available to all
 /// HTTP API calls.
-pub struct ServerState {
-    pub db_executor: Addr<dispatcher::DBExecutor>,
-}
+pub struct ServerState;
```

Comment:
I kept `ServerState` around even though it's empty now because presumably it's going to be populated with some kind of db property in due course.